microservice/src/test/java/de/gsi/microservice/concepts/majordomo/MajordomoBroker.java

```diff
@@ -0,0 +1,648 @@
+                MajordomoProtocol.sendClientMessage(client.socket, MdpClientCommand.C_UNKNOWN, clientMessage.senderID, clientMessage.serviceNameBytes, "501".getBytes(StandardCharsets.UTF_8));
+                return;
+            }
+            // queue new client message RBAC-priority-based
+            service.putPrioritisedMessage(clientMessage);
+
+            // dispatch service
+            if (service.isInternal) {
+                final MdpClientMessage msg = service.getNextPrioritisedMessage();
+                if (msg == null) {
+                    // should be thrown only with VM '-ea' enabled -- assert noisily since this a (rare|design) library error
+                    assert false : "getNextPrioritisedMessage should not be null";
+                    return;
+                }
+                MajordomoProtocol.sendWorkerMessage(service.internalDispatchSocket, MdpWorkerCommand.W_REQUEST, null, msg.senderID, msg.payload);
+            } else {
+                //dispatch(requireService(clientMessage.serviceName, clientMessage.serviceNameBytes));
+                dispatch(service);
+            }
+        });
+    }
+
+    /**
+     * Process message sent to us by a worker.
+     */
+    protected void processWorker(final Socket receiveSocket, final MdpWorkerMessage msg) {
+        boolean workerReady = receiveSocket == internalServiceSocket || workers.containsKey(msg.senderIdHex);
+        final Worker worker; // = requireWorker(receiveSocket, msg.senderID, msg.senderIdHex);
+        switch (msg.command) {
+        case W_READY:
+            worker = requireWorker(receiveSocket, msg.senderID, msg.senderIdHex);
+            // Not first mdpWorkerCommand in session || Reserved service name
+            if (workerReady || Arrays.equals(INTERNAL_SERVICE_PREFIX_BYTES, 0, 3, msg.senderID, 0, 3))
+                deleteWorker(worker, true);
+            else {
+                // Attach worker to service and mark as idle
+                worker.service = requireService(msg.serviceName, msg.serviceNameBytes);
+                workerWaiting(worker);
+            }
+            break;
+        case W_REPLY:
+            if (workerReady) {
+                final Client client = clients.get(msg.clientSourceName);
+                if (client == null || client.socket == null) {
+                    break;
+                }
+                if (client.protocol == MdpSubProtocol.C_CLIENT) { // OpenCMW
+                    MajordomoProtocol.sendClientMessage(client.socket, MdpClientCommand.C_UNKNOWN, msg.clientSourceID, msg.serviceNameBytes, msg.payload);
+                } else {
+                    // TODO: add other branches for:
+                    // * CmwLight
+                    // * REST/JSON
+                    // * REST/HTML
+                    throw new IllegalStateException("Unexpected value: " + client.protocol);
+                }
+                if (receiveSocket != internalServiceSocket) {
```

Comment:
Codacy found an issue: [Use equals() to compare object references.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)